speculation-rules.bs

```diff
@@ -368,7 +377,12 @@ add the following step
   1. If |input|["`target_hint`"] [=map/exists=]:
     1. If |input|["`target_hint`"] is not a [=valid navigable target name or keyword=], then return null.
     1. Set |targetHint| to |input|["`target_hint`"].
-  1. Return a [=speculation rule=] with [=speculation rule/URLs=] |urls|, [=speculation rule/predicate=] |predicate|, [=speculation rule/requirements=] |requirements|, and [=speculation rule/target navigable name hint=] |targetHint|.
+  1. Let |referrerPolicy| be the empty string.
+  1. If |input|["`referrer_policy`"] [=map/exists=]:
+    1. If |input|["`referrer_policy`"] is not a [=referrer policy=], then return null.
+    1. If |input|["`referrer_policy`"] is the empty string, then return null.
```

Comment:
domenic, offline you mentioned using the empty string to indicate falling back. Is that internally to the spec, or developer facing as well?

For example, should
```
{ "prefetch": [{
  "source": "list",
  "urls": ["https://example.com/"],
  "referrer_policy": ""
 }]
}
```
be treated as if the "referrer_policy" key were omitted, or should this be treated as an invalid rule? The implementation currently does the latter.

Reply:
I find it most elegant if we allow the empty string here. Everywhere else on the platform, it means to fall back. So I think it should here, too. It also gives us a nice default value in the parsed data model, such that the empty string is the same as omitted.

Finally, it's possible this could be useful for authors generating speculation rules in JavaScript (or a server-side language). The following code:

```js
const rule = {
  source: "list",
  urls: computeURLs(),
  referrer_policy: computeReferrerPolicy() // might return ""
};
```

is a bit simpler than


```js
const rule = {
  source: "list",
  urls: computeURLs(),
  referrer_policy: computeReferrerPolicy() // might return ""
};

if (rule.referrer_policy === "") {
  delete rule.referrer_policy;
}
```

Reply:
Makes sense, thanks. Updated this to allow the empty string.